verifyCopyrightHeaders.ps1

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+$VerbosePreference = 'Continue'
+$ErrorActionPreference = 'Stop'
+
+$copyrightHeader  = 'Copyright (c) Microsoft Corporation. All rights reserved.'
+$mitLicenseHeader = 'Licensed under the MIT License.'
+$xmlHeader        = '<?xml'
+$shellHeader        = '#!/bin/bash'
+
+$resultFile = "verifyCopyrightHeadersResult.log"
+Write-Verbose "Generating $resultFile" 
+New-Item $resultFile -Force | Out-Null
+
+$excludedDirs =  @(
+    # generated
+    '.vs'
+    '.vscode'
+    'bin'
+    'Debug'
+    'obj'
+    'package'
+    'Release'
+
+    # external
+    'atl'
+    'empty'
+    'winsdk'
+
+    # test-related
+    'InstrEngineTests\\Baselines'
+    'InstrEngineTests\\TestScripts'
+)
+
+$excludedExtensions = @(
+    '.filters'
+    '.htm'
+    '.log'
+    '.md'
+    '.rgs'
+    '.sln'
+    '.snk'
+    '.txt'
+    '.user'
+)
+
+$excludedFiles = @(
+    # external
+    'corprof.h'
+    'dbgmsg.h'
+
```

Comment:
Do we not have a separate folder for externals?

Reply:
Not at the moment. I can move these into one update the refs in a separate checkin.